src/rgw/rgw_rados.cc

```diff
@@ -8579,6 +8579,14 @@ int RGWRados::cls_bucket_list_ordered(const DoutPrefixProvider *dpp,
   return 0;
 }
 
+static int parse_index_hash_source(const string& oid_wo_ns, string *index_hash_source) {
+  std::size_t found = oid_wo_ns.rfind('.');
+  if (found == std::string::npos || found < 1) return -EINVAL;
+  found = oid_wo_ns.rfind('.', found -1);
```

Comment:
"-1" looks like negative one, where adding a space "- 1" clarifies that you're subtracting one.